motoko/minimal-counter-dapp/README.md

```diff
@@ -0,0 +1,212 @@
+- Implement backend functions in the frontend
+- Deploy the canister smart contract locally
+- Test backend with Candid UI and command line using DFX, and test frontend in browser
+
+## Installation
+This example project can be cloned, installed and deployed locally, for learning and testing purposes. The instructions are based on running the example on either macOS or Linux, but when using WSL2 on Windows, the instructions will be the same.
+
+### Prerequisites
+The example project requires the following installed:
+
+- git
+- Node.js
+- DFX
+
+git and Node can be installed from various package managers. DFX can be installed following the instructions [here](https://smartcontracts.org/docs/quickstart/local-quickstart.html#download-and-install).
+
+### Install
+Install the example dapp project:
+
+```bash
+$ git clone https://github.com/dfinity/examples
+$ cd motoko/minimal-dapp
+$ npm install
+```
+
+The project folder will then look like this:
+
+![Project Files](README_images/project_files.png)
+
+
+## Documentation
+The three main parts of the example dapp are the backend, the Candid interface and the frontend. This example project is based on the default project, which is created when running the `dfx new project_name` command, but most of the default project code is replaced to create the counter functionality in this project.
+
+### Motoko backend
+The backend functions are located in the `src/minimal_dapp/main.mo` Motoko file. The backend stores the counter value, and has functions to get, increment and reset the counter value.
+
+
+#### Counter variable
+Three functions are created to make the counter work: `count()`, `getCount()` and `reset()`. The current counter value is stored as a number in the actor.
+
+
+```javascript
+actor {
+    var counter : Nat = 0;
+}
+```
+
+#### count()
+The `count()` function increments the counter variable. This function is invoked when the user clicks the button on the frontend, or when the function is called through the Candid interface.
+
+```javascript
+public func count() : async Nat {
+    counter += 1;
+    return counter;
+};
+```
+
+The function returns the incremented counter variable.
+
+#### getCount()
+The `getCount()` function returns the current counter value.
+
+```javascript
+public query func getCount() : async Nat {
+    return counter;
+};
+```
+
+#### reset()
+The `reset()` function resets the counter value to 0 and returns the value.
+
+```javascript
+public func reset() : async Nat {
+    counter := 0;
+    return counter;
+};
+```
+
+### Candid interface
+The Candid interface is automatically created, and it has a convenient UI, which provides an easy, user-friendly way to test the backend. The UI is also automatically generated, and the canister ID can be retrieved from the `dfx canister id <canister_name>` command.
+
+```bash
+$ dfx canister id __Candid_UI
+r7inp-6aaaa-aaaaa-aaabq-cai
+```
+
+**http://<candid_canister_id>.localhost:8000/?id=<backend_canister_id>**
```

Comment:
Following this readme from the beginning, the canister hasn't been created yet, so `dfx canister id` fails.  Can we move this section to follow the **Testing** section?

Otherwise the reader will see this:
```
Cannot find canister id. Please issue 'dfx canister create __Candid_UI'.
```